### PR TITLE
FSF-2956 Feat/cone heuristic

### PR DESCRIPTION
**Pull request by Davide64-dev**

### CODE REVIEW

File: src/perception/src/cone_evaluator/distance_predict.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    double coneDistance = cluster.getCentroid().norm();
+
+    if (coneDistance <= 0) return 0;
+
+    double vertical_component = CONE_HEIGHT / (2.0 * coneDistance *
+                        tan(vertical_ang_res * M_PI / 180.0 / 2));
+    double horizontal_component = CONE_WIDTH / (2.0 * coneDistance *
+                        tan(horizontal_ang_res * M_PI / 180.0 / 2));
+
+    double n_points_prediction = 0.5 * vertical_component * horizontal_component;
```
Comment (marhcouto) [inline src/perception/src/cone_evaluator/distance_predict.cpp:22]: This is the number of points you think you should be receiving? The only think that concerns me here is that I don't see any part of the equations above taking the shape of the cone into account

File: src/perception/test/distance_validator_test.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  for (int i = 0; i < 4; i++) {
+    point_cloud->push_back(pcl::PointXYZI(25.0, 0.0, 0.0, 0.1));
+  }
+
+  Cluster cluster = Cluster(point_cloud);
+  auto coneEvaluator = new DistancePredict(0.33, 0.2);
+
+  double coneConfidence = coneEvaluator->evaluateCluster(cluster);
+
+  ASSERT_LE(coneConfidence, 1.0);
```
Comment (marhcouto) [inline src/perception/test/distance_validator_test.cpp:75]: These assertions are only checking for utter absurdities, shouldn't they be a little tighter?

**Review by marhcouto — CHANGES_REQUESTED** [review]
The part of the messages is in another PR?

**Review by PedroRomao3 — APPROVED** [review]
seems ok, besides previous notes

**Review by marhcouto — APPROVED** [review]

**Review by marhcouto — APPROVED** [review]

**Pull request merged**